core/src/main/java/com/amplifyframework/auth/AuthCategoryBehavior.java

```diff
@@ -0,0 +1,146 @@
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void confirmSignUp(
+            @NonNull String username,
+            @NonNull String code,
+            @NonNull Consumer<AuthSignUpResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * If the user's code expires or they just missed it, this method can
+     * be used to send them a new one.
+     * @param username The username of the account to resend the code to
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void resendSignUpCode(
+            @NonNull String username,
+            @NonNull Consumer<AuthSignUpResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Basic authentication to the app with a username and password or, if custom auth is setup,
+     * you can send null for those and the necessary authentication details in the options object.
+     * @param username This can be a normal username, email/phone if that's setup, or null if custom auth is enabled
+     * @param password User's password for normal signup, null if custom auth or passwordless configurations are setup
+     * @param options Advanced options such as a map of auth information for custom auth
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void signIn(
+            @Nullable String username,
+            @Nullable String password,
+            @NonNull AuthSignInOptions options,
+            @NonNull Consumer<AuthSignInResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Basic authentication to the app with a username and password.
+     * @param username This can be a normal username or email/phone if that's setup
+     * @param password User's password
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void signIn(
+            @Nullable String username,
+            @Nullable String password,
+            @NonNull Consumer<AuthSignInResult> onSuccess,
+            @NonNull Consumer<AuthException> onError);
+
+    /**
+     * Launch a pre-built sign in UI flow. You should also put the {@link #handleSignInWithUIResponse(Intent)} method in
+     * your activity's onResume method to capture the response which comes back from the UI flow.
+     * @param callingActivity The activity in your app you are calling this from
+     * @param onSuccess Success callback
+     * @param onError Error callback
+     */
+    void signInWithUI(
+            @NonNull Activity callingActivity,
+            @NonNull Consumer<String> onSuccess,
+            @NonNull Consumer<AmplifyException> onError);
+
+    /**
+     * Handles the response which comes back from {@link #signInWithUI(Activity, Consumer, Consumer)}.
+     * @param intent The app activity's intent
+     */
+    void handleSignInWithUIResponse(Intent intent);
```

Comment:
What happens on error?

Reply:
The triggering function above, signInWithUI, gets that error in it's error listener

Reply:
> The triggering function above, signInWithUI, gets that error in it's error listener

Should add this to doco